happi/cli.py

```diff
@@ -873,6 +885,151 @@ def no_create_channel(*args, **kwargs):
+    whether naturally or via assert calls.  These functions take a single
+    happi.SearchResult as an positional argument and returns None if
+    successful.
+
+    To import additional checks, provide a file with your check function
+    and a list named ``checks`` containing the desired functions.
+    """
+    logger.debug('Starting audit block')
+
+    # if a file is provided, make its functions available
+    if ext_file:
+        fp = Path(ext_file)
+        sys.path.insert(1, str(fp.parent))
+        ext_module = importlib.import_module(fp.stem)
+        ext_checks = getattr(ext_module, 'checks')
+        checks.extend(ext_checks)
+
+    # List checks subcommand
+    if list_checks:
+        check_pt = prettytable.PrettyTable(field_names=['name',
+                                                        'description'])
+        check_pt.hrules = prettytable.ALL
+        check_pt.align['description'] = 'l'
+        for chk in checks:
+            check_pt.add_row([chk.__name__, inspect.cleandoc(chk.__doc__)])
+        print(check_pt)
+        return
+
+    # gather selected checks
+    if check_choices:
+        check_list = []
+        for check_name in check_choices:
+            # check if provided check name is a substring of any checks
+            matches = [fn for fn in checks if check_name in fn.__name__]
+            if len(matches) != 1:
+                raise click.BadParameter(
+                    f'provided check name ({check_name}) must match only'
+                    f'one check.  Matches: ({[ch.__name__ for ch in matches]})'
+                )
+            check_list.append(matches[0])
+    else:
+        # take all checks
+        check_list = checks
+
+    client: happi.Client = ctx.obj
+
+    results = search_parser(client, use_glob, search_criteria)
+    logger.info(f'found {len(results)} items to verify')
+    logger.info(f'running checks: {[f.__name__ for f in check_list]}')
+
+    test_results = {'name': [], 'success': [], 'check': [], 'msg': []}
+    f = io.StringIO()
+    for i, res in enumerate(results):
+        if not (names_only or show_json):
+            print(f'checking device #: {i}', end='\r')
+        # Capture stdout, stderr for this audit
+        with redirect_stderr(f), redirect_stdout(f):
+            for check_fn in check_list:
+                success, check, msg = verify_result(res, check_fn)
+                test_results['name'].append(res.item.name)
+                test_results['success'].append(success)
+                test_results['check'].append(check)
+                test_results['msg'].append(msg)
+
+    unique_fails = set(test_results['name'][i]
+                       for i in range(len(test_results['name']))
+                       if not test_results['success'][i])
+    # print outs
+    if names_only:
+        click.echo(' '.join(unique_fails))
+    elif show_json:
+        final_dict = {'audited': len(results),
+                      'falures': len(unique_fails)}
```

Comment:
I just noticed the additional `--json` feature. People and their side-channel requests 🙄 

Looks good! Just one little typo:
```suggestion
                      'failures': len(unique_fails)}
```

Reply:
I used to be good at spelling sigh